docs: improve DR volumes (#1147)
longhorn/website: content/docs/1.10.0/snapshots-and-backups/setup-disaster-recovery-volumes.md

---
title: Disaster Recovery (DR) Volumes
weight: 4
---

> In the dynamic world of containerized applications, ensuring data resilience is crucial. A Longhorn Disaster Recovery (DR) volume is a special volume designed to handle this by storing data in a backup cluster, safeguarding you in case your main cluster goes down. It's a standby volume of a primary volume, created in a secondary Kubernetes cluster from backups.

Reviewer: ```suggestion
Ensuring data resilience is important when working with containerized applications.
A **Longhorn Disaster Recovery (DR) volume** is a special type of volume designed to maintain a standby copy of data in a secondary Kubernetes cluster. It is created from backups of a primary volume and kept in sync to enable rapid recovery if the primary cluster becomes unavailable.
```

The main purpose of a DR volume is to provide a near-real-time copy of your data in a geographically separate location, enabling rapid service restoration in the event of a complete site outage. This approach minimizes both data loss, known as the Recovery Point Objective (RPO), and the time required to bring applications back online in the recovery cluster, known as the Recovery Time Objective (RTO).

## How It Works

The functionality of DR volumes relies on asynchronous replication through a shared backup store.

- **Shared Backup Target**

    Your primary and secondary Kubernetes clusters must be configured to use the exact same external backup target (e.g., an S3-compatible object store or an NFS share).

- **Incremental Backup and Restore**

    A DR volume is created from an existing backup. It then continuously polls the backup target for newer backups from the source volume and restores them incrementally. The `Last Backup` field in the UI shows the most recent backup that has been restored.

>     To keep the DR volume updated, you can configure recurring jobs on the source volume to perform regular incremental backups. These recurring backups provide the DR volume with new backups to restore, helping ensure minimal data loss in the event of a disaster.

Reviewer: ```suggestion
    To keep the DR volume updated, configure recurring jobs on the source volume to perform regular incremental backups. These recurring backups provide the DR volume with new backups to restore, helping ensure minimal data loss in the event of a disaster.
```

### Using `kubectl` Command

1. **Get the Backup URL**: First, copy the full URL of the source backup from the Backup page in the Longhorn UI. The format of this URL will depend on your configured backup target (e.g., S3, NFS).

2. **Create a YAML Manifest**: Create a file (e.g., dr-volume.yaml) with the following content. Be sure to replace the placeholder URL and adjust the name, size, accessMode, etc to match your source volume. In this file, the `standby: true` field defines the volume as a DR standby volume.

    ```yaml
    apiVersion: longhorn.io/v1beta2
    kind: Volume
    metadata:
      name: example-dr-volume
      namespace: longhorn-system
    spec:
      size: "2147483648"
      accessMode: rwo
      numberOfReplicas: 3
      fromBackup: "nfs://longhorn-nfs-server.example.com:/opt/backupstore?backup=backup-b69a1249e97f4a27&volume=pvc-33509786-92d7-427c-9b5a-b6d61d56b063"
      # This flag is essential to create a standby volume
      Standby: true
    ```

3. **Apply the Manifest**: Apply the manifest to your **secondary cluster** to create the volume.

## Activating a DR Volume

> When a failover is necessary, you must activate the DR volume to make it writable.
> 
> Longhorn supports activation under these conditions:

Reviewer: ```suggestion
When a failover is necessary, you must activate the DR volume to make it writable.  
Longhorn supports activation under the following conditions:
```

- The volume is healthy, indicating that all replicas are in a healthy state.
- The volume is degraded (some replicas have failed), but only if the global setting [`Allow Volume Creation with Degraded Availability`](../../references/settings/#allow-volume-creation-with-degraded-availability) is enabled.

>     > **Note**: When the setting `Allow Volume Creation with Degraded Availability` is disabled, attempting to activate a degraded DR volume will cause the volume to become stuck in the attached state. However, after enabling the setting, the DR volume will be activated and converted into a normal volume, remaining in the detached state.

Reviewer: ```suggestion

> **Note**: When the setting `Allow Volume Creation with Degraded Availability` is disabled, attempting to activate a degraded DR volume will cause the volume to become stuck in the `Attached` state.  
> After enabling the setting, the DR volume will be activated and converted into a normal volume, remaining in the `Detached` state.
```

### Using Longhorn UI

1. Go to the `Volume` page in the Longhorn UI of your secondary cluster.
2. Select the DR volume you want to activate.

> 3. Click the `Activate Disaster Recovery Volume` button in dropdown menu of Operation.

Reviewer: ```suggestion
3. Click the `Activate Disaster Recovery Volume` button in the **Operation** dropdown menu.
```
